Formula/gnat.rb

```diff
@@ -0,0 +1,152 @@
+class Gnat < Formula
+  desc "GNU New York University Ada Trans"
+  homepage "https://gcc.gnu.org/"
+  url "https://ftp.gnu.org/gnu/gcc/gcc-11.1.0/gcc-11.1.0.tar.xz"
+  mirror "https://ftpmirror.gnu.org/gcc/gcc-11.1.0/gcc-11.1.0.tar.xz"
+  sha256 "4c4a6fb8a8396059241c2e674b85b351c26a5d678274007f076957afa1cc9ddf"
+  license "GPL-3.0-or-later" => { with: "GCC-exception-3.1" }
+  head "https://gcc.gnu.org/git/gcc.git"
+
+  livecheck do
+    url "https://ftp.gnu.org/gnu/gcc/gcc-11.1.0"
+    regex(%r{href=.*?gcc[._-]v?(\d+(?:\.\d+)+)(?:/?["' >]|\.t)}i)
+  end
+
+  # The bottles are built on systems with the CLT installed, and do not work
+  # out of the box on Xcode-only systems due to an incorrect sysroot.
+  pour_bottle? do
+    on_macos do
+      reason "The bottle needs the Xcode CLT to be installed."
+      satisfy { MacOS::CLT.installed? }
+    end
+  end
+
+  depends_on arch: :x86_64
+
+  depends_on "gmp"
+  depends_on "isl"
+  depends_on "libmpc"
+  depends_on "mpfr"
+  depends_on "zstd"
+
+  uses_from_macos "libiconv"
+  uses_from_macos "zlib"
+
+  on_linux do
+    depends_on "binutils"
+  end
+
+  conflicts_with "gcc", because: "both install gcc"
```

Comment:
Can we remove the files that conflict with `gcc`, so that this doesn't conflict with `gcc`? That's basically what `libgccjit` does.

Reply:
GNAT invokes `gcc` to compile ada source code so that is inevitable.

Reply:
Can it be persuaded to invoke `gcc` from the `gcc` formula?

Reply:
If use `gcc` from `gcc` formula
```
% gnat make --GCC=gcc-11 hello.adb
gcc-11 -c hello.adb
gcc-11: error: hello.adb: Ada compiler not installed on this system
gnatmake: "hello.adb" compilation error
```

Reply:
[`specs.h`](https://github.com/gcc-mirror/gcc/blob/ea34e2edd3d7ab245d1f57a1487c10587f324ec6/gcc/gcc.c#L1493) in `gcc` formula doesn't include `gcc/ada/gcc-interface/lang-specs.h`